build-tools/packages/build-tools/README.md

```diff
@@ -130,58 +119,98 @@ fluid-build                   # build
-    and they should be updated together -->
-
-```txt
-Usage: fluid-layer-check <options>
-Options:
-     --dot <path>     Generate *.dot for GraphViz
-     --info <path>    Path to the layer graph json file
-     --md [<path>]    Generate PACKAGES.md file for human consumption at path relative to repo root (default: repo root)
-     --defroot <path> Default root directory of the Fluid repo if infer failed (default: env _FLUID_DEFAULT_ROOT_)
-     --root <path>    Root directory of the Fluid repo (default: env _FLUID_ROOT_)
-     --timer          Measure elapsed time of each step
-     --logtime        Display the current time on every status message for logging
-  -v --verbose        Verbose messages
+### Task and dependency definition
+
+`fluid-build` uses task and dependency definitions to construct a build graph. It is used to determine which task and
+the order to run in. The default definitions are located in at the root `fluidBuild.config.cjs` file under the `tasks` property.
+This definitions applies to all packages in the repo. Script tasks and dependencies specified in this default definitions
+doesn't have to appear on every package and will be ignored if it is not found.
+
+The task definitions is an object with task names as keys, the task dependencies and config to define the action of the task.
+For more details, see the definition `TaskDefinitionsOnDisk` in [./src/common/fluidTaskDefinitions.ts](./src/common/fluidTaskDefinitions.ts)
+
+For example:
+
+```js
+module.exports = {
+   tasks: {
+      "build": {
+         dependsOn: [ "tsc", "build:test" ]           // Depends on the `tsc` nad `build:esnext` task in the same package
+         script: false                                // Don't trigger a npm script
+      }
+      "tsc": [ "^tsc"],                               // Depends on `tsc` task of all of the dependent packages
+                                                      // (if the task exists)
+      "build:test": [ "tsc" ],                        // Depends on `tsc` task
+      "clean": {
+         before: ["*"],                               // If the task "clean" is specified, it runs before all other task
+      }
+      "prettier": [],                                 // No dependent tasks
+   }
+}
 ```
 
-By default, without any options, fluid-layer-check checks the dependencies in Fluid Framework packages are layered properly and warn about if they are not. The property layering is defined in `data/layerInfo.json` file.
-
-With --dot &lt;path&gt; argument, it will generate the dependency graph in the [dot](https://graphviz.gitlab.io/_pages/doc/info/lang.html) format and [GraphViz](https://graphviz.org/) can be used to generate visual representation of our packages.
-
-## Details `fluid-build`
+Each package can be augmented the tasks definition by adding task definitions under `fluidBuild.tasks` in `package.json`.
+
+For example:
+
+```json
```

Comment:
```suggestion
```json5
```

Reply:
Actually, package.json isn't json5.  So I don't think I want to have it here.